Common/Source/SaveLoadTask/LoadCupTask.cpp

```diff
@@ -233,252 +250,544 @@ class CupObsZoneUpdater {
+                if (_tcscmp(pToken, _T("Options")) == 0) {
+                    while ((pToken = strsep_r(NULL, TEXT(","), &pWClast)) != NULL) {
+                        if (_tcsstr(pToken, _T("NoStart=")) == pToken) {
+                            // Opening of start line
+                            PGNumberOfGates = 1;
+                            StrToTime(pToken + 8, &PGOpenTimeH, &PGOpenTimeM);
+                        } else if (_tcsstr(pToken, _T("TaskTime=")) == pToken) {
+                            // Designated Time for the task
+                            // TODO :
+                        } else if (_tcsstr(pToken, _T("WpDis=")) == pToken) {
+                            // Task distance calculation. False = use fixes, True = use waypoints
+                            // TODO :
+                        } else if (_tcsstr(pToken, _T("NearDis=")) == pToken) {
+                            // Distance tolerance
+                            // TODO :
+                        } else if (_tcsstr(pToken, _T("NearAlt=")) == pToken) {
+                            // Altitude tolerance
+                            // TODO :
+                        } else if (_tcsstr(pToken, _T("MinDis=")) == pToken) {
+                            // Uncompleted leg.
+                            // False = calculate maximum distance from last observation zone.
+                            // TODO :
+                        } else if (_tcsstr(pToken, _T("RandomOrder=")) == pToken) {
+                            // if true, then Random order of waypoints is checked
+                            // TODO :
+                        } else if (_tcsstr(pToken, _T("MaxPts=")) == pToken) {
+                            // Maximum number of points
+                            // TODO :
+                        } else if (_tcsstr(pToken, _T("BeforePts=")) == pToken) {
+                            // Number of mandatory waypoints at the beginning. 1 means start line only, two means
+                            //      start line plus first point in task sequence (Task line).
+                            // TODO :
+                        } else if (_tcsstr(pToken, _T("AfterPts=")) == pToken) {
+                            // Number of mandatory waypoints at the end. 1 means finish line only, two means finish line
+                            //      and one point before finish in task sequence (Task line).
+                            // TODO :
+                        } else if (_tcsstr(pToken, _T("Bonus=")) == pToken) {
+                            // Bonus for crossing the finish line
+                            // TODO :
+                        }
+                    }
+                } else if (_tcsstr(pToken, _T("ObsZone=")) == pToken) {
+                    TCHAR *sz = NULL;
+                    CupObsZoneUpdater TmpZone;
+                    TmpZone.mIdx = _tcstol(pToken + 8, &sz, 10);
+                    if (TmpZone.mIdx < MAXSTARTPOINTS) {
```

Comment:
must be "if (TmpZone.mIdx < MAXSTARTPOINTS) {"

or that revert an older fix : 
https://github.com/LK8000/LK8000/commit/028b0ed3fad9a51af49ee887970acf3cb0e0cc29

Reply:
Yes, sorry!
done!